File: client/src/lib/invoke/saveConfigToFile.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Config } from "@bindings";

export function saveConfigToFile(config: Config, path_string: string): ResultAsync<void, string> {
    return ResultAsync.fromPromise(
        invoke("save_config_to_file", { config, path_string }),
        (error) => `Failed to save config to file: ${error}`
    );
}

File: client/src-tauri/src/commands/save_config_to_file.rs
use crate::config::Config;

#[tauri::command]
pub async fn save_config_to_file(
    config: Config,
    path_string: String,
) -> Result<(), String> {
    tauri::async_runtime::spawn_blocking(move || {
        Config::to_file(&config, &path_string).map_err(|err| err.to_string())
    })
    .await
    .map_err(|err| format!("Failed to save config to file: {err}"))?
}
